pymeasure/instruments/hp/hp3478A.py

```diff
@@ -0,0 +1,389 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+
+import logging
+
+from pymeasure.instruments import Instrument
+
+
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+
+class HP3478A(Instrument):
+    """ Represents the Hewlett Packard 3748A 5 1/2 digit multimeter
+    and provides a high-level interface for interacting
+    with the instrument.
+
+    As this unit predates SCPI some tricks are required to get this working
+    """
+
+    DIGITS={3: "N3",
+            4: "N4",
+            5: "N5"
+            }
+
+    AZ={"off": "Z0",
+        "on": "Z1"
+        }
+
+    MODES={"DCV": "F1",
+           "ACV": "F2",
+           "R2W": "F3",
+           "R4W": "F4",
+           "DCI": "F5",
+           "ACI": "F6",
+           "Rext": "F7"
+           }
+
+    RANGES={"DCV": {3E-2: "R-2",3E-1: "R-1",3: "R0",30: "R1",300:"R2","auto": "RA"},
+            "ACV": {3E-1: "R-1",3: "R0",30: "R1" ,300: "R2","auto": "RA"},
+            "R2W": {30: "R1",300: "R2", 3E3: "R3",3E4: "R4",3E5: "R5", 3E6: "R6",
+                    3E7: "R7","auto": "RA"},
+            "R4W": {30: "R1",300: "R2", 3E3: "R3",3E4: "R4",3E5: "R5", 3E6: "R6",
+                    3E7: "R7","auto": "RA"},
+            "DCI": {3E-1: "R-1",3: "R0","auto": "RA"},
+            "ACI": {3E-1: "R-1",3: "R0","auto": "RA"},
+            "Rext": {3E6: "R7","auto": "RA"}
+            }
+
+    TRIGGERS={
+        "auto": "T1",
+        "internal": "T1",
+        "external": "T2",
+        "single": "T3",
+        "hold": "T4",
+        "fast": "T5"
+        }
+
+    def __init__(self, resourceName, **kwargs):
+        super(HP3478A, self).__init__(
+            resourceName,
+            "Hewlett-Packard HP3478A",
+            includeSCPI=False,
+            send_end=True,
+            read_termination="\r\n",
+            **kwargs
+        )
+
+
+    def error_status(self):
+        """
+        get the Error-status register value
+
+        Returns
+        -------
+        estatus : int
+            error status register value (confirm with manual for detail)
+
+        """
+        estatus=self.ask('E')
+        return estatus
+
```

Comment:
Is it possible to implement the commands using the `Instrument.control/setting/measurement` class methods ?
For example, does the following works ?
```
    error_status = Instrument.measurement(
        "E",
        """ This property returns the Error-status register value""",
    )

```

Reply:
This might work, need to dig into this a bit more

Reply:
found that this works
'    error_status = Instrument.measurement(
        "E",
        """Checks the error status register""",
        get_process=(lambda x: int(x)),
        )
'

actually I am thinking to change the name to check_errors